packages/ckeditor5-ui/src/colorpicker/colorpickerview.ts

```diff
@@ -114,7 +114,13 @@ export default class ColorPickerView extends View {
 		} );
 
 		this.on( 'change:_hexColor', () => {
-			this.picker.setAttribute( 'color', this._hexColor );
+			const input = this.hexInputRow!.children.get( 1 ) as LabeledFieldView<InputTextView>;
+
+			// Should update color in color picker when change was caused by input or when
+			// color picker is not visible.
+			if ( input.isFocused || this.picker.offsetParent === null ) {
```

Comment:
I think we could simplify this check and also make it more intuitive (looking at you, `this.picker.offsetParent === null`). We want to set the color attribute always except the case when the picker sliders are focused. So the following should do the job:

```suggestion
			if ( document.activeElement !== this.picker ) {
```

Please verify it on your end and update the `should not set any color directly to color-picker component when it visible` unit test (it should be much simpler with simply setting the active element instead of creating some nested fixture).